src/Azure/Azure.Quantum.Client/JobManagement/Workspace.cs

```diff
@@ -267,5 +313,8 @@ public async Task<string> GetSasUriAsync(string containerName, string blobName =
                 jobId,
                 inner);
         }
+
+        private static string NormalizeLocation(string location) =>
+            string.Concat(location.Where(c => !char.IsWhiteSpace(c))).ToLower();
```

Comment:
Why is whitespace not allowed but other invalid URI characters are?

Reply:
Removing whitespace characters was part of the feature specification and I will incorporate Ryan's suggestion of additionally checking it is a valid location through Uri.CheckHostName.

Reply:
Is there a link to the feature specification?

Reply:
I added a validator to the location which passes the normalized value provided by the user to the `Uri.CheckHostName` method to verify it.

I sent a link to you regarding the details on the whitespace :)